src/pnpq/apt/protocol.py

```diff
@@ -688,6 +688,117 @@ class AptMessage_MGMSG_MOD_IDENTIFY(AptMessageHeaderOnlyChanIdent):
     message_id = AptMessageId.MGMSG_MOD_IDENTIFY
 
 
+@dataclass(frozen=True, kw_only=True)
+class AptMessage_MGMSG_MOT_GET_POSCOUNTER(AptMessageWithData):
+    data_length: ClassVar[int] = 6
+    message_id: ClassVar[AptMessageId] = AptMessageId.MGMSG_MOT_GET_POSCOUNTER
+    message_struct: ClassVar[Struct] = Struct(
+        f"{AptMessageWithData.header_struct_str}{ATS.WORD}{ATS.LONG}"
+    )
+
+    chan_ident: ChanIdent
+    position: int
+
+    @classmethod
+    def from_bytes(cls, raw: bytes) -> "AptMessage_MGMSG_MOT_GET_POSCOUNTER":
+        (
+            message_id,
+            data_length,
+            destination,
+            source,
+            chan_ident,
+            position,
+        ) = cls.message_struct.unpack(raw)
+
+        if message_id != cls.message_id:
+            raise ValueError(
+                f"Expected message ID {cls.message_id.value}, but received {message_id} instead. Full raw data was {raw!r}"
+            )
+        if data_length != cls.data_length:
+            raise ValueError(
+                f"Expected data packet length {cls.data_length}, but received {data_length} instead. Full raw data was {raw!r}"
+            )
+        if destination & 0x80 != 0x80:
+            raise ValueError(
+                f"Expected the destination's highest bit to be 1, indicating that a data packet follows, but it was 0. Full raw data was {raw!r}"
+            )
+
+        return AptMessage_MGMSG_MOT_GET_POSCOUNTER(
+            destination=Address(destination & 0x7F),
+            source=Address(source),
+            chan_ident=ChanIdent(chan_ident),
+            position=position,
+        )
+
+    def to_bytes(self) -> bytes:
+        return self.message_struct.pack(
+            self.message_id,
+            self.data_length,
+            self.destination_serialization,
+            self.source,
+            self.chan_ident,
+            self.position,
+        )
+
+
+@dataclass(frozen=True, kw_only=True)
+class AptMessage_MGMSG_MOT_SET_POSCOUNTER(AptMessageWithData):
```

Comment:
Since the GET and SET command structures are essentially the same, we should avoid duplicating all this code and instead create a parent class that both can inherit from, containing all the serialization and deserialization logic. See AptMessageWithDataMotorStatus for an example of this.